infra/services/aiSearch.bicep

```diff
@@ -0,0 +1,156 @@
+targetScope = 'resourceGroup'
+
+@minLength(4)
+@maxLength(24)
+@description('The base resource name.')
+param baseName string = resourceGroup().name
+
+@description('The location of the resource. By default, this is the same as the resource group.')
+param location string = resourceGroup().location
+
+@description('The tenant ID to which the application and resources belong.')
+param tenantId string = '72f988bf-86f1-41af-91ab-2d7cd011db47'
+
+@description('The client OID to grant access to test resources.')
+param testApplicationOid string
+
+// Reference to storage account - we need this for RBAC assignment
+resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing = {
+  name: baseName
+}
+
+// Create searchdocs container in the storage account for document storage
+resource searchDocsContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2022-09-01' = {
+  name: '${storageAccount.name}/default/searchdocs'
+  properties: {
+    publicAccess: 'None'
+  }
+}
+
+// Azure AI Search service
+resource search 'Microsoft.Search/searchServices@2025-02-01-preview' = {
+  name: baseName
+  location: location
+  sku: {
+    name: 'free'
+  }
+  properties: {
```

Comment:
Assuming the tests calling to Azure Search will use an Entra identity and not an API key, you'll need to enable RBAC auth here, add a property like this:

```json
"authOptions": {
  "aadOrApiKey": {
    "aadAuthFailureMode": "http401WithBearerChallenge"
  }
}
```

If you also want to disable api key-based auth you can add a separate property `"disableLocalAuth": false`

Reply:
thanks.  added